### ISSUE TITLE
Connection to Echo

### ISSUE THREAD

**Issue by Niklas04** [Dragon863/EchoCLI#28]
Hi,

I want to root my Echo Dot (3. Generation) - however, I'm not sure about connecting it via USB.

I soldered a USB-Cable to PCS - as described here:
[https://www.briandorey.com/post/echo-dot-3rd-gen-digging-deeper#base-board](https://www.briandorey.com/post/echo-dot-3rd-gen-digging-deeper#base-board)

When connecting the Echo Dot and powering it up, I see the following in the kernel-log:
Dec 28 15:57:49 Ubuntu-Niklas kernel: [  378.382402] usb 3-1: new low-speed USB device number 2 using xhci_hcd
Dec 28 15:57:49 Ubuntu-Niklas kernel: [  378.518466] usb 3-1: device descriptor read/64, error -71
Dec 28 15:57:49 Ubuntu-Niklas kernel: [  378.762519] usb 3-1: device descriptor read/64, error -71
Dec 28 15:57:49 Ubuntu-Niklas kernel: [  378.998393] usb 3-1: new low-speed USB device number 3 using xhci_hcd
Dec 28 15:57:49 Ubuntu-Niklas kernel: [  379.134496] usb 3-1: device descriptor read/64, error -71
Dec 28 15:57:50 Ubuntu-Niklas kernel: [  379.378474] usb 3-1: device descriptor read/64, error -71
Dec 28 15:57:50 Ubuntu-Niklas kernel: [  379.486479] usb usb3-port1: attempt power cycle
Dec 28 15:57:50 Ubuntu-Niklas kernel: [  379.898370] usb 3-1: new low-speed USB device number 4 using xhci_hcd
Dec 28 15:57:50 Ubuntu-Niklas kernel: [  379.899259] usb 3-1: Device not responding to setup address.
Dec 28 15:57:50 Ubuntu-Niklas kernel: [  380.107245] usb 3-1: Device not responding to setup address.
Dec 28 15:57:51 Ubuntu-Niklas kernel: [  380.314350] usb 3-1: device not accepting address 4, error -71
Dec 28 15:57:51 Ubuntu-Niklas kernel: [  380.442361] usb 3-1: new low-speed USB device number 5 using xhci_hcd
Dec 28 15:57:51 Ubuntu-Niklas kernel: [  380.443245] usb 3-1: Device not responding to setup address.
Dec 28 15:57:51 Ubuntu-Niklas kernel: [  380.651247] usb 3-1: Device not responding to setup address.
Dec 28 15:57:51 Ubuntu-Niklas kernel: [  380.858349] usb 3-1: device not accepting address 5, error -71
Dec 28 15:57:51 Ubuntu-Niklas kernel: [  380.858491] usb usb3-port1: unable to enumerate USB device

Can someone confirm, if this is the expected behavior?
I would expect something like mentioned here:
[https://gitlab.com/phodina/echo-debug-gen3#usb-enumeration](https://gitlab.com/phodina/echo-debug-gen3#usb-enumeration)

Tried it also with a Echo (2. Generation) - same results in kernel-log.

Would be glad for any help

**Comment by Dragon863** [comment]
No, that's not right. The fact there are multiple errors with trying to send to the device would imply that the data connection to the echo isn't working in the TX direction - maybe check the soldering/wiring?

**Comment by Niklas04** [comment]
Thank you for the clarification.
It was indeed D+ and D- switched.

Leaving the pinning here - in case anyone needs it.
For the Echo Dot 3rd Gen, the pinning is as follows:
TM3: GND
TM4: D-
TM5: D+

For the Echo 2nd Gen, the pinning is as follows:
TP13: GND
TP14: D-
TP15: D+

Interestingly the Echo 2nd Gen also uses the MT8163 and the same type of eMMC as the Echo Dot 2nd Gen.
So I will try first to root the Echo 2nd Gen first.
Do you think I can just give it a try with your scripts or are there maybe any adjustments needed for the Echo 2nd Gen?
From a short look into your code, there seem to be some checks implemented, when the LK. Don't know if fos-flags are the same though.

**Comment by Dragon863** [comment]
Fos_flags will likely be the same, but I’m not sure about anything else. It’s pretty likely that it could work, but as I’ve never tried it there is a risk of bricking the echo. If you do decide to go forward with it, let me know how it goes!

**Comment by Niklas04** [comment]
Great news - it actually did work :D
I now have a rooted Echo 2nd Gen :D

The only real difference in progress was the disassembly.
The Echo 2nd Gen has three PCBs instead of two: The Microphone/LED-Board, the Main-Board (with the MT8163 and the eMMC), and the Amp-Board (with the USB-Interface, Power- and AUX-Plug and the connections for the speakers)
![PXL_20250103_104533208](https://github.com/user-attachments/assets/505aa318-9e07-4c51-8c40-ec5069a7a090)
I needed a Torx 8 and a Torx 10 for disassembly

The inside of the RF cage is pretty similar to the Echo Dot 2nd Gen. You also have to short C52 to go into BootRom Mode.
![PXL_20250103_104542249](https://github.com/user-attachments/assets/47228d92-d76d-4dd6-88f9-b9659affb961)

I did run the EchoCLI scripts without any modifications. Also, the Fos_flags seem to be the same as the Echo Dot 2nd Gen - I used the recommended settings for now - ADB is working.

By the way, the Codename of the Echo 2nd Gen is Puffin.

Thanks for your support!

**Comment by Dragon863** [comment]
Awesome! interestingly there are a bunch of references to Puffin in the dot's software too, so I guess they're actually very similar in software and hardware. Thanks for the info!
